sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.search.annotation.FieldIgnore;
+import com.azure.search.annotation.FieldProperty;
+import com.azure.search.annotation.SearchableFieldProperty;
+import com.azure.search.annotation.SimpleFieldProperty;
+import com.azure.search.documents.models.AnalyzerName;
+import com.azure.search.documents.models.DataType;
+import com.azure.search.documents.models.Field;
+import com.azure.search.documents.models.GeoPoint;
+
+import java.lang.annotation.Annotation;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Helper to build search field
+ */
+public class FieldBuilder {
+    /**
+     * Creates a collection of <see cref="Field"/> objects corresponding to
+     * the properties of the type supplied.
+     *
+     * @param modelClass The class type for which fields will be created, based on its properties.
+     * @return A collection of fields.
+     */
+    public static <T> List<Field> build(Class<T> modelClass) {
+        ClientLogger logger = new ClientLogger(FieldBuilder.class);
+        java.lang.reflect.Field[] fields = modelClass.getFields();
+        List<Field> searchFieldList = new ArrayList<>();
+        for (java.lang.reflect.Field field: fields) {
+            Annotation[] annotations = field.getAnnotations();
+            for (Annotation annotation : annotations) {
+                if (annotation instanceof FieldIgnore) {
+                    continue;
+                }
+                Field searchField = new Field();
+                searchField.setName(field.getName());
+                searchField.setType(covertToDataType(field, logger));
+                if (annotation instanceof SimpleFieldProperty) {
+                    buildSimpleField(searchField, (SimpleFieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                } else if (annotation instanceof SearchableFieldProperty) {
+                    buildSearchableField(searchField, (SearchableFieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                } else if (annotation instanceof FieldProperty) {
+                    buildField(searchField, (FieldProperty) annotation);
+                    searchFieldList.add(searchField);
+                }
+                logger.info("The property {} is not a search field.", field.getName());
```

Comment:
What about `ComplexField`? Just not implemented yet?

Reply:
This is for the "static" case where we're reflecting over a type, right? In that case, you shouldn't need an attribute to identify complex types -- the type of the property/accessor should indicate that (by not being a primitive, string, or geo-point or collection of those).

Reply:
That's implying that non-attributed fields should be reflected and turned into fields. That doesn't seem to follow what track 1 did, though I only took a cursory glance. If we should reflect non-attributed fields, then that means all attributes' properties' values would be default, correct?

Reply:
@heaths Track 1 reflected all public properties, even those without attributes. One reason is that complex fields don't require any additional attributes (because they already have a name and type). As for defaults, in Track 1 it was the client's defaults (everything disabled by default). In Track 2, that would be a safe default to have, but I'm not sure if that's different than the defaults for explicitly-constructed `SearchFields`.

Reply:
Okay, so long as we're agreed that since we're following convenience classes' defaults for these convenience attributes (or lack thereof), then we will have a consistent experience. E.g. if a string field/property has no attributes, it will be equivalent to a `SimpleField` of "Edm.String" and no other traits enabled, including that `IsHidden` is `false` so it will appear in results.

Reply:
Thank for comments! Will treat non-attributed field to complex field if it is non-primitive.